docs/internet-identity-spec.adoc

```diff
@@ -197,6 +197,8 @@ type DeviceData = record {
   pubkey : DeviceKey;
   alias : text;
   credential_id : opt CredentialId;
+  purpose : { #recovery; #authentication };
+  type : { #unknown; #platform; #crossPlatform; #seedPhrase };
```

Comment:
`type` is a keyword in Rust. Can we name this `key_type` or something like that?